src/systemd_ctypes/inotify.py

```diff
@@ -28,15 +28,24 @@ class inotify_event(ctypes.Structure):
         ('len', ctypes.c_uint32),
     )
 
+    # Cache for dynamically created classes to prevent memory leaks
+    _len_struct_classes: ClassVar[Dict[int, Type[Any]]] = {}
+
     @property
     def name(self) -> Optional[bytes]:
         if self.len == 0:
             return None
 
-        class event_with_name(ctypes.Structure):
-            _fields_ = (*inotify_event._fields_, ('name', ctypes.c_char * self.len))
+        # cache the classes,
+        if self.len not in self._len_struct_classes:
```

Comment:
This starts to feel like a separate function wrapped in `@lru_cache` with an extremely large maximum.  (note: no `@cache` in 3.6).

Reply:
Right, I told my AI minion to do that, and it looks reasonable.